lms/validation/_launch_params.py

```diff
@@ -68,3 +70,28 @@ def _verify_oauth_1(self, _data):
             self._launch_verifier.verify()
         except LTILaunchVerificationError as err:
             raise marshmallow.ValidationError("Invalid OAuth 1 signature.") from err
+
+
+class URLConfiguredLaunchParamsSchema(PyramidRequestSchema):
+    """
+    Schema for an "URL-configured" Basic LTI Launch.
+
+    An URL-configured launch is one where the content URL is provided by a "url"
+    launch param.
+    """
+
+    url = marshmallow.fields.Str(required=True)
+
+    @marshmallow.post_load
+    def _decode_url(self, _data):  # pylint:disable=no-self-use
```

Comment:
I do wonder about globally disabling the no-self-use rule

Reply:
Hmm ... I _do_ think it provides useful feedback generally, in that I think it is preferable to use a plain function if a method doesn't need `this`, or it may indicate a bug if `self` is not used. The exception is for functions that need to be a method for use with some external library or if you're overriding an inherited method and just want to return a constant.